modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -963,7 +964,9 @@ else if (!deleteRelationshipObjectField) {
 			objectField1 -> !objectField1.isMetadata());
 
 		if (objectDefinition.isApproved() && (objectFields.size() == 1)) {
-			throw new RequiredObjectFieldException();
+			throw new RequiredObjectFieldRelationshipException.
+				RequiredObjectFieldRelationship(
+					objectDefinition.getLabel(), "RelationshipName");
```

Comment:
Hey @larissacribeiro 

You can use `objectDefinition.getShortName()` instead `objectDefinition.getLabel()`. I believe it will solve your problem.